test/fixtures/generator/bwb.xcodeproj/xcshareddata/xcschemes/generator.xcscheme

```diff
@@ -57,7 +57,7 @@
       </PreActions>
       <BuildActionEntries>
          <BuildActionEntry
-            buildForTesting = "NO"
+            buildForTesting = "YES"
```

Comment:
This is wrong. The previous way is the expected default. It should only default to all on for ones that don't have a test action or launch action implying the buildFor.

Reply:
I changed the scheme to replicate the missing `build_for` value. Previously, we only specified a value for `archiving`. The rest of the values were unspecified. Hence, the merging resulted in `buildForTesting` being `NO`. The updated scheme does not specify a `build_for` value. So, I set the default to all enabled, based upon this line from our previous conversation:
> buildActionInfo can define which ones (in BUILD files), but defaults to all (like today)

[Comment from FR](https://github.com/buildbuddy-io/rules_xcodeproj/issues/573#issuecomment-1203310073)

I think that the disconnect is that the `tools/generator` scheme does not need to specify any build targets. I added it as an integration test of sorts. Perhaps, it would make more sense to remove it from `tools/generator` and create some schemes in `examples/multiplatform` or `examples/ios_app` that had some interesting combinations.

For now, it might be best for me to put the `tools/generator` scheme back to its previous state. What do you think?

For clients that really want to specify a build target in a scheme, I think the all enabled may still make sense. The only reason that they should add the build target is if the target is not referenced anywhere else in the scheme.

Reply:
I think it changed since I made that comment. I can use an empty `build_for()` and all the values will be `unspecified`, which should be the same as not passing a `build_for()` at all (these two are semantically the same to me, and shouldn't differ). And if they are all `unspecified`, I believe we turn them all on in the generator it nothing else ends up implying something different, right?

Reply:
The main issue is the semantic difference between none and an empty `build_for()`. I'm fine with it defaulting to all enabled, as it makes sense.

Reply:
> My understanding was that if a client specifies a build target, they are trying to customize the build_for or they just want all of the phases enabled.

That does end up being the case most of the time, but I think if they don't specify `build_for`, then we don't know their intent, and if they want to customize that they can.

But maybe that's the wrong way of looking at it. Maybe it's exactly as you say, and then the only change I would request is if someone uses `build_for()` without specifying anything (all are `undefined`) we turn them all to `enabled`. This way both not specifying it and specifying an empty one end up behaving the same.

@erikkerber @jpsim How were you using this? What was the expected outcome? I assume it was listing a target that wasn't listed in a `launch_action` or `test_action`, like grouping a top-level framework target with some unassociated app and test targets (in which case the default would end up building for all actions)?

Reply:
One thing I thought of: if someone starts with a `build_target` and later adds a test that depends on it; it would be surprising for the build_for to change.

So, you/I have talked myself back on that part. Let's just make an empty `build_for()` an error to resolve that semantic difference issue.

Reply:
@cgrindel Don't make any more changes. I'll take this PR as it stands for now. I have some ideas how to tweak this further, but I can take that on myself as I write some docs for this feature.